components/nexusoperations/executors_test.go

```diff
@@ -326,6 +331,23 @@ func TestProcessInvocationTask(t *testing.T) {
 				require.Equal(t, 0, len(events))
 			},
 		},
+		{
+			name:                  "invocation timeout by ScheduleToCloseTimeout",
+			requestTimeout:        time.Hour,
+			schedToCloseTimeout:   time.Microsecond,
+			destinationDown:       true,
+			expectedMetricOutcome: "request-timeout",
+			onStartOperation: func(ctx context.Context, service, operation string, input *nexus.LazyValue, options nexus.StartOperationOptions) (nexus.HandlerStartOperationResult[any], error) {
+				time.Sleep(time.Millisecond * 100) //nolint:forbidigo // Allow time.Sleep for timeout tests
+				return &nexus.HandlerStartOperationResultAsync{OperationID: "op-id"}, nil
+			},
+			checkOutcome: func(t *testing.T, op nexusoperations.Operation, events []*historypb.HistoryEvent) {
+				require.Equal(t, enumsspb.NEXUS_OPERATION_STATE_BACKING_OFF, op.State())
+				require.NotNil(t, op.LastAttemptFailure.GetApplicationFailureInfo())
+				require.Regexp(t, "Post \"http://localhost:\\d+/service/operation\\?callback=http%3A%2F%2Flocalhost%2Fcallback\": context deadline exceeded", op.LastAttemptFailure.Message)
```

Comment:
Hmm... we should probably not show the HTTP endpoint in the error message. But that's out of scope for this PR.